### PR TITLE
Implement IpDataSender

### PR DESCRIPTION
**Pull request by bonewell**
One moment: I tried to send (using pcap_inject) string to TUN on QNX. There were not errors, but tcpdump didn't catch any packets.  I hope it is because ip packets were not correct.
On Linux tcpdump catches packet but says about bad ip addresses.

### CODE REVIEW

File: src/components/ivdcm_module/src/ivdcm.cc [hunk 1]
```diff
@@ -67,17 +68,22 @@ Ivdcm::Ivdcm()
   // TODO(KKolodiy) workaround for reading profile,
   // possible it is fixed in fresh version of OpenSDL
   profile::Profile::instance()->config_file_name("smartDeviceLink.ini");
+  proxy_ = new IvdcmProxy(this);
```
Comment (DKlimenko) [inline src/components/ivdcm_module/src/ivdcm.cc]: nothrow

**Comment by VladSemenyuk** [comment]
reviewed

File: src/components/ivdcm_module/src/ivdcm.cc (hunk 1 above)
Comment (bonewell) [inline src/components/ivdcm_module/src/ivdcm.cc]: I'll fix with implementation of IpDataReceiver.

File: src/components/ivdcm_module/src/ivdcm_proxy.cc [hunk 2]
```diff
@@ -44,9 +48,56 @@ namespace ivdcm_module {
 
 CREATE_LOGGERPTR_GLOBAL(logger_, "IVDCM")
 
+struct ShowPacketInfo {
+  explicit ShowPacketInfo(const std::vector<uint8_t>* packet)
+    : packet_(packet) {
+  }
+  const std::vector<uint8_t>* packet_;
+};
+std::ostream& operator << (std::ostream& output, const ShowPacketInfo& info) {
```
Comment (DKlimenko) [inline src/components/ivdcm_module/src/ivdcm_proxy.cc]: Is it possible not to use streams?
As I mentioned before streams could raise exceptions if you don't disable them explicitly.

**Comment by DKlimenko** [comment]
reviewed

File: src/components/ivdcm_module/src/ivdcm_proxy.cc (hunk 2 above)
Comment (bonewell) [inline src/components/ivdcm_module/src/ivdcm_proxy.cc]: Could you give link on standard where describe this one?

**Pull request closed without merging**